server/covmanager/templates/collections/report.html

```diff
@@ -138,24 +210,67 @@
       this.reverse = (this.sortKey === sortKey) ? !this.reverse : false
       this.sortKey = sortKey
     },
+    browse: function(id) {
+      window.open(`../collections/${id}/browse/`, '_blank').focus()
+    },
     summary: function(id) {
       window.open(`../collections/${id}/summary/`, '_blank').focus()
     },
-    collection_click_handler: function(id) {
+    report_click_handler: function(id) {
       var self = this
-      let idx = self.selected_collections.indexOf(id)
+      let idx = self.selected_reports.indexOf(id)
       let target = $("#" + id)
 
       if (idx < 0) {
-        self.selected_collections.push(id)
-        if (target)
-          target.toggleClass("collection-selected", true)
+        self.selected_reports.push(id)
       } else {
-        self.selected_collections.splice(idx, 1)
-        if (target)
-          target.toggleClass("collection-selected", false)
+        self.selected_reports.splice(idx, 1)
       }
     },
+    publish: function (report) {
+      let url = URLS.reports_api + "update/" + report.id + "/"
+      let data = { "public" : !report.public }
+      this.update(url, data)
+    },
+    promote: function (report) {
```

Comment:
Does this refresh the view when you promote/demote, or just update the affected row?

ie. If you accidentally promote weekly->monthly, you need to click again twice to reset back to weekly. Is this going to refresh the page between each click?

Reply:
The data is loaded in the background and the view updated, so this is a data refresh but not a full page refresh. This is because the `update` function calls `fetch`.

Reply:
Perfect, thanks!